product-service/src/functions/catalogBatchProcess/handler.ts

```diff
@@ -0,0 +1,85 @@
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
+import { PublishCommand, SNSClient } from '@aws-sdk/client-sns';
+import { DynamoDBDocumentClient, TransactWriteCommand } from '@aws-sdk/lib-dynamodb';
+import { v4 as uuidv4 } from 'uuid';
+
+import { ProductPostBody } from '../../models/product';
+
+import type { Handler, SQSEvent } from 'aws-lambda';
+
+const dynamoDb = new DynamoDBClient({ region: process.env.AWS_REGION });
+const ddbClient = DynamoDBDocumentClient.from(dynamoDb);
+const sns = new SNSClient({ region: process.env.AWS_REGION });
+
+const catalogBatchProcess: Handler<SQSEvent> = async (event) => {
```

Comment:
It would be better to have a flatter structure of the file - divide the logic into appropriate functions.
At the moment, the code is hard to read.